File: mns-reactor/src/main/java/cz/fb/manaus/reactor/categorizer/AbstractPriceCountCategorizer.java
package cz.fb.manaus.reactor.categorizer;

import cz.fb.manaus.core.model.SettledBet;
import cz.fb.manaus.core.model.Side;

public abstract class AbstractPriceCountCategorizer extends AbstractCountCategorizer {

    private final Side side;

    public AbstractPriceCountCategorizer(String prefix, int maxCount, Side side) {
        super(prefix, maxCount);
        this.side = side;
    }

    @Override
    protected int getCount(SettledBet bet) {
        return bet.getBetAction().getMarketPrices().getRunnerPrices(bet.getSelectionId())
                .getHomogeneous(side).getPrices().size();
    }
}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/categorizer/AbstractPriceCountCategorizer.kt
package cz.fb.manaus.reactor.categorizer

import cz.fb.manaus.core.model.RealizedBet
import cz.fb.manaus.core.model.Side
import cz.fb.manaus.core.model.getRunnerPrices

abstract class AbstractPriceCountCategorizer(prefix: String, maxCount: Int, private val side: Side) : AbstractCountCategorizer(prefix, maxCount) {

    override fun getCount(bet: RealizedBet): Int {
        return getRunnerPrices(bet.betAction.runnerPrices, bet.settledBet.selectionId)
                .getHomogeneous(side).prices.size
    }
}
